### ISSUE TITLE
Is this project still running?

### ISSUE THREAD

**Issue by Shideravan** [NGEmu/NGEmu#1]
I never saw a true N-Gage emulator out there. Is this project still running?
Did you achieve any progress on emulation?
Maybe I can help you with something :)

**Comment by tambry** [comment]
We're still at work reverse-engineering various parts of the N-Gage. Unfortunately the knowledge of the boot process is lacking. We'd also be interested in the contents of the ROM chips on the N-Gage, but unfortunately no one on the team has any hardware expertise.  
I personally am still working on a project that I hope will also become basis for further N-Gage emulation efforts. I think LLE might be worth researching at this time, as it would eventually allow us to understand the OS much more easily through a debugger.

Feel free to join #ngemu on Freenode if you are interested in helping out.

**Comment by inferno986return** [comment]
My technical knowledge of the NGAGE is limited aside from the obvious of course. However there's got to be a way to promote this to the community. The console had quite a few exclusives I'm interested in emulating.

**Comment by jan-kleks** [comment]
@tambry Have you tried to post to [/r/ReverseEngineering/](https://www.reddit.com/r/ReverseEngineering/) or [ASSEMBlergames](https://assemblergames.com) regarding these ROM chips?

If you would have to share any news regarding the project or look for assistance in the future, you can post to /r/emulation, [I have recently told people there about NGEmu](https://www.reddit.com/r/emulation/comments/7kgl9u/ngemu_ngage_emulator_basic_cpu_emulation_and_some/).

Regards and keep up good work!

**Comment by tambry** [comment]
@jan-kleks No, I have not. I doubt anyone else has bothered to dump them before. Although if there are people willing to do that themselves on there, then I think it'd definitely be worth a shot. I also remember that we have the schematics for the whole N-Gage and the corresponding docs and manuals - this should make dumping much easier, if anyone wants to go ahead with it. The pinout for those ROM chips should be on there too. They should be available somewhere in the Symbian Archive or the wiki.  
If no one else takes it up, then I'll eventually find the time (likely during the summer) to learn more about hardware and hopefully dump them.

Rest assured, I'm still interested in N-Gage emulation. I'll continue the research and will hopefully eventually produce an emulator, even if it takes a long time.

**Comment by jan-kleks** [comment]
@tambry And one more question, I understand that you focus on the Nokia N-Gage, but could your work be beneficial when it comes to the emulation of Symbian games in the broader sense?

**Comment by tambry** [comment]
@jan-kleks As my personal goal is LLE, then work towards accomplishing that for a single device (in this case the N-Gage) may be useful for LLE'ing other similar Symbian OS devices. Chances are many of them use similar or even same components.

A bit more detailed explanation. At a lower level, Symbian OS has a HAL (hardware-abstraction layer) and a concept of drivers. If the emulator were able to emulate a given hardware component well enough for the driver to interface with it without problems, then another device using the same (or a similar) driver and the same hardware component should need no extra work done to support a given hardware component. This would drastically reduce the work needed to support additional Symbian OS devices, once the emulation for the first few devices have been implemented.

**Comment by huckleberrypie** [comment]
Yeah, emulating the N-Gage would also mean emulating pretty much every S60v1 and S60v2 platform device, as the N-Gage is essentially a Symbian platform device with more RAM and a dedicated audio DSP. Looking forward to this as I do miss playing Symbian games anyway.

**Comment by minakmostoles** [comment]
I think that is a great project. I follow you and if I can contribute something, I will.

**Comment by jan-kleks** [comment]
[EKA2L1: A Symbian OS emulator](https://www.reddit.com/r/emulation/comments/8aotzq/eka2l1_a_symbian_os_emulator/) For anyone interested... It's good that more and more people are starting to work on these Symbian platforms.
